lottie-swift/src/Public/Animation/AnimationView.swift

```diff
@@ -764,14 +764,17 @@ final public class AnimationView: LottieView {
       case .pause:
         removeCurrentAnimation()
       case .pauseAndRestore:
-        currentContext.closure.ignoreDelegate = true
+        currentContext.closure.ignoreDelegate = !waitingToPlayAimation
         removeCurrentAnimation()
         /// Keep the stale context around for when the app enters the foreground.
         self.animationContext = currentContext
```

Comment:
The completion block of an animation _should_ be called after restored as-is. The context is stored, along with its completion block, here. When the app resumes the animation and its completion block is started again and should be called.
If that is not happening then there is a larger bug in the animation context. This solution is more of a bandaid that will further mask the real bug.